cpp/src/plasma/CMakeLists.txt

```diff
@@ -189,17 +186,17 @@ if(ARROW_PLASMA_JAVA_CLIENT)
                           plasma_shared
                           ${PLASMA_LINK_LIBS}
                           "-undefined dynamic_lookup"
-                          -Wl,-force_load,${FLATBUFFERS_STATIC_LIB}
-                          ${FLATBUFFERS_STATIC_LIB}
+                          -Wl,-force_load,flatbuffers::flatbuffers
+                          flatbuffers::flatbuffers
                           ${PTHREAD_LIBRARY})
   else(APPLE)
     target_link_libraries(plasma_java
                           plasma_shared
                           ${PLASMA_LINK_LIBS}
                           -Wl,--whole-archive
-                          ${FLATBUFFERS_STATIC_LIB}
+                          flatbuffers::flatbuffers
                           -Wl,--no-whole-archive
-                          ${FLATBUFFERS_STATIC_LIB}
+                          flatbuffers::flatbuffers
                           ${PTHREAD_LIBRARY})
```

Comment:
@pcmoritz @robertnishihara do you know why `libflatbuffers.a` is required in these libraries?